File: 27-Remove-Element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        for( int i = 0; i < nums.size(); i++ )
        {
            if( nums[i] == val )
                nums[i] = 1001;
        }

        sort(nums.begin(), nums.end());

        int ans = 0;
        for( int i = 0; i < nums.size(); i++ )
        {
            if( nums[i] != 1001 )
                ans++;
        }

        return ans;
    }
};

File: 27-Remove-Element.go
func removeElement(nums []int, val int) int {
    for i := 0; i < len(nums); i++ {
        if nums[i] == val {
            nums[i] = 1001
        }
    }

    sort.Ints(nums);
    ans := 0
    for i := 0; i < len(nums); i++ {
        if nums[i] != 1001 {
            ans++
        }
    }

    return ans;
}